### DISCUSSION TITLE
How to take back the ownership of a component from the world?

### DISCUSSION THREAD

**Discussion by navneetankur** [bevyengine/bevy#13116] [Q&A] [answered]
Is it possible to take back the ownership of component from world? Or replace it with another without Drop-ing.
Something like what std::mem::take or swap does for vectors?

**Comment by SkiFire13** [comment, marked as answer]
You can use `world.entity_mut(your_entity_id)` to get a `EntityWorldMut` and then call `.take::<YourComponentType>()` on it to get ownership of the component.

> without Drop-ing.

Please be careful with that since not dropping some value probably means leaking memory, which in the long run is going to make your program run out of memory and crash.

**Reply by navneetankur** [reply]
Thanks.

> Please be careful with that since not dropping some value probably means leaking memory

How? It should automatically Drop-ed when the new owner get dropped right?

**Reply by SkiFire13** [reply]
> How? It should automatically Drop-ed when the new owner get dropped right?

Well, if it gets a new owner of course it won't be dropped until the owner gets dropped. I just assumed that since you said "without Drop-ing" you meant for it to not get a new owner, but I guess I misinterpreted it.

**Comment by alice-i-cecile** [comment]
You should also be able to use the various remove component APIs on the world: those will return ownership of the returned values.

**Reply by navneetankur** [reply]
I wasn't able to find anything like remove_component in [World](https://docs.rs/bevy_ecs/latest/bevy_ecs/world/struct.World.html). There is a World::remove_resource, but I assumed it won't work for Component.

**Reply by alice-i-cecile** [reply]
[`EntityWorldMut::remove`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.EntityWorldMut.html#method.remove) is the API you want :) Accessed via [`World::get_entity_mut`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.World.html#method.get_entity_mut).

**Reply by navneetankur** [reply]
Thanks,
Though it doesn't return back the Component. :)

pub fn remove<T>(&mut self) -> &mut EntityWorldMut<'w>
